rules/type-declaration/tests/Rector/ClassMethod/AddArrayParamDocTypeRector/Fixture/data_provider.php.inc

```diff
@@ -4,38 +4,18 @@ namespace Rector\TypeDeclaration\Tests\Rector\ClassMethod\AddArrayParamDocTypeRe
 
 use PHPUnit\Framework\TestCase;
 
+/**
+ * Though provideData returns an array. dataProviders work take the values 2 levels deep inside this array.
+ * This code shouldn't be altered by AddArrayParamDocTypeRector as the doc type should not be an array.
+ * It would be a string.
+ * See: https://phpunit.readthedocs.io/en/9.3/writing-tests-for-phpunit.html#data-providers
+ */
 class DataProvider extends TestCase
 {
     /**
-     * @dataProvider provideData()
+     * @dataProvider provideData
      */
-    public function testSomething(array $names)
-    {
-    }
-
-    public function provideData(): array
-    {
-        return [
-            ['Bob', 'Josh']
-        ];
-    }
-}
-
-?>
------
-<?php
-
-namespace Rector\TypeDeclaration\Tests\Rector\ClassMethod\AddArrayParamDocTypeRector\Fixture;
-
-use PHPUnit\Framework\TestCase;
-
-class DataProvider extends TestCase
-{
-    /**
-     * @dataProvider provideData()
-     * @param string[] $names
-     */
-    public function testSomething(array $names)
+    public function testSomething( $name1, $name2 )
```

Comment:
This should not be skipped, but fixed to correct types:

```suggestion
    public function testSomething(string $name1, string $name2)
```

Reply:
This test only has the ParamDocType rule applied though, so only the phpdocs should be touched if anything as far as I can understand?
And It also looks like thi rule should only apply to arrays?

Reply:
Ah yes, I've missed that. You're right